### PR TITLE
fix tf.nn.{conv2d,convolution} substitution

### PR DESCRIPTION
**Pull request by irenaby**
## Pull Request Description:
Fixed tf conv substitution to handle attrs with default values that were not passed explicitly, and convert tf-specific format to keras compatible values.  

## Checklist before requesting a review:
- [ ] I set the appropriate labels on the pull request.
- [ ] I have added/updated the release note draft (if necessary).
- [ ] I have updated the documentation to reflect my changes (if necessary).
- [ ] All function and files are well documented. 
- [ ] All function and classes have type hints.
- [ ] There is a licenses in all file.
- [ ] The function and variable names are informative. 
- [ ] I have checked for code duplications.
- [ ] I have added new unittest (if necessary).

### CODE REVIEW

File: model_compression_toolkit/core/keras/graph_substitutions/substitutions/conv_funcs_to_layer.py [hunk 1]
```diff
@@ -136,42 +137,68 @@ def substitute(self,
...
+            return graph
+        conv_fw_attr[DILATION_RATE] = dilations
+
         if b is None:
             conv_fw_attr[USE_BIAS] = False
         else:
             weights[BIAS] = b
 
+        data_format = conv_func_node.op_call_kwargs.get(DATA_FORMAT, 'NHWC')
+        conv_fw_attr['data_format'] = {'NHWC': 'channels_last', 'NCHW': 'channels_first'}[data_format]
```
Comment (elad-c) [inline model_compression_toolkit/core/keras/graph_substitutions/substitutions/conv_funcs_to_layer.py:166]: use constants from the keras constants

File: model_compression_toolkit/core/keras/graph_substitutions/substitutions/conv_funcs_to_layer.py [hunk 2]
```diff
@@ -136,42 +137,68 @@ def substitute(self,
...
+        Args:
+            node: node
+            key: param key
+
+        Returns:
+            Parsed value or None if non-standard.
+        """
+        v = node.op_call_kwargs.get(key)
+        if v is None:
+            return 1, 1
```
Comment (elad-c) [inline model_compression_toolkit/core/keras/graph_substitutions/substitutions/conv_funcs_to_layer.py:190]: this way you assume the defaults. why not return None?

File: tests/keras_tests/feature_networks_tests/feature_networks/linear_collapsing_test.py [hunk 3]
```diff
@@ -54,6 +54,8 @@ def compare(self, quantized_model, float_model, input_x=None, quantization_info=
         y = float_model.predict(input_x)
         y_hat = quantized_model.predict(input_x)
         self.unit_test.assertTrue(y.shape == y_hat.shape, msg=f'out shape is not as expected!')
+        # FIXME this doesn't test anything, the number of quantized convs in the network is exactly 0. Even if it
```
Comment (elad-c) [inline tests/keras_tests/feature_networks_tests/feature_networks/linear_collapsing_test.py:57]: then why not remove it?

File: tests/keras_tests/feature_networks_tests/feature_networks/linear_collapsing_test.py [hunk 4]
```diff
@@ -107,9 +110,44 @@ def create_networks(self):
...
-        for layer in quantized_model.layers:
-            if type(layer) == layers.Conv2D:
-                self.unit_test.assertTrue(len(layer.weights) == 2,msg=f'fail Bias should appear in weights!!')
+        convs = [l for l in quantized_model.layers if isinstance(l, layers.Conv2D)]
+        self.unit_test.assertTrue(len(convs) == 1)
+        for layer in convs:
+            self.unit_test.assertTrue(len(layer.weights) == 2,msg=f'fail Bias should appear in weights!!')
+
+
+class FuncConv2DCollapsingTest(FourConv2DCollapsingTest):
```
Comment (elad-c) [inline tests/keras_tests/feature_networks_tests/feature_networks/linear_collapsing_test.py:119]: These tests seem redunant, as the there's no difference between these tests and the ones for Conv2D layer. wht not just test the substitution of these layers to Conv2D?

**Review by elad-c — COMMENTED** [review]
Add PR description

File: model_compression_toolkit/core/keras/graph_substitutions/substitutions/conv_funcs_to_layer.py (hunk 2 above)
Comment (irenaby) [reply model_compression_toolkit/core/keras/graph_substitutions/substitutions/conv_funcs_to_layer.py:190]: That's intentional. None wouldn't do, we need to fill in an explicit default. This method is specific to tf stride & dilation

**Review by elad-c — APPROVED** [review]
